sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -112,7 +112,25 @@ object JdbcUtils extends Logging {
    */
   def insertStatement(conn: Connection, table: String, rddSchema: StructType, dialect: JdbcDialect)
       : PreparedStatement = {
-    val columns = rddSchema.fields.map(x => dialect.quoteIdentifier(x.name)).mkString(",")
+    // Use database column names instead of RDD schema column names
+    val tableSchemaQuery = conn.prepareStatement(dialect.getSchemaQuery(table))
```

Comment:
I think the table schema won't change during inserting all data. You ask table schema for every insert statement now. Can we do this once in caller side (i.e., `savePartition`) and reuse the schema then?